client/cloudevents/types/types.go

```diff
@@ -0,0 +1,97 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	ClusterAll = ""
+	SourceAll  = ""
+)
+
+const (
+	SubResourceSpec   = "spec"
+	SubResourceStatus = "status"
+)
+
+const (
+	ResyncRequestAction  = "resync_request"
+	ResyncResponseAction = "resync_response"
+)
+
+const (
+	ExtensionResourceID        = "resourceid"
+	ExtensionResourceVersion   = "resourceversion"
+	ExtensionDeletionTimestamp = "deletiontimestamp"
+	ExtensionClusterName       = "clustername"
+	ExtensionOriginalSource    = "originalsource"
+)
+
+type ResourceAction string
+
+const (
+	Added          ResourceAction = "ADDED"
+	Modified       ResourceAction = "MODIFIED"
+	StatusModified ResourceAction = "STATUSMODIFIED"
+	Deleted        ResourceAction = "DELETED"
+)
+
+// ListOptions is the query options for listing the resource objects from the source/agent.
+type ListOptions struct {
+	// Source use the cluster name to restrict the list of returned objects by their cluster name.
+	// Defaults to all clusters.
+	ClusterName string
+
+	// Agent use the source ID to restrict the list of returned objects by their source ID.
+	// Defaults to all sources.
+	Source string
+}
+
+// GroupVersionResource identifies the type of the resource included in the cloud event payload.
+type GroupVersionResource struct {
```

Comment:
we cannot use schema.GroupVersionResource directly?

Reply:
using the schema.GroupVersionResource